hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -274,6 +325,42 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
     }
   }
 
+  // GPIO command interfaces
+  for (auto i = 0u; i < info_.gpios.size(); i++)
+  {
+    const auto & gpio = info_.gpios[i];
+    for (const auto & interface : gpio.command_interfaces)
+    {
+      if (!get_interface(
+            gpio.name, gpio_interfaces_, interface.name, i, gpio_commands_, command_interfaces))
+      {
+        throw std::runtime_error(
+          "Interface is not found in the standard nor other list. "
+          "This should never happen!");
+      }
+    }
+  }
+
+  // Fake gpio command interfaces
+  if (fake_gpio_command_interfaces_)
+  {
+    for (auto i = 0u; i < info_.gpios.size(); i++)
+    {
+      const auto & gpio = info_.gpios[i];
+      for (const auto & interface : gpio.state_interfaces)
+      {
+        if (!get_interface(
+              gpio.name, gpio_interfaces_, interface.name, i, gpio_fake_commands_,
```

Comment:
If you are using this, arn't you doubling "real command" and "fake" interfaces?

In the fake interfaces there should be only indices where there are no "real commands". Or for the simplicity, we can simply create "all" interfaces as fake if option is activated and then ignore real command interfaces.

What do you think?

Reply:
As discussed, in case of 'fake_gpio_command_interfaces_' true, it will use fake command interfaces for all the command interfaces, else it will use real command interfaces. From the user point of view, there will be no effect.